### ISSUE TITLE
Using argopy to fetch Argo data ?

### ISSUE THREAD

**Issue by gmaze** [ArgoCanada/bgcArgoDMQC#37]
Hi,
Great work ! Awesome to see more python tools to work with Argo data.

This year we started the development of an Argo python library to fetch Argo data _from any source_: https://github.com/euroargodev/argopy
We're ok with T/S/P data and our next step is to fetch BGC data (https://github.com/euroargodev/argopy/issues/22)
Argopy supports caching, parallelization and manages seemingly for users access to erddap, a local ftp, argovis (and soon the online ftp), 

At some point in 2021, argopy will be used internally on the [new OWC implementation in python](https://github.com/euroargodev/argodmqc_owc), 

Basically, our philosophy is that python tools for Argo should be delegating data fetching to a 3rd party library, namely argopy

Do you think you would have an interest in implementing data fetching with argopy once we'll have full BGC data support ?

**Comment by cgrdn** [comment]
Hi @gmaze - yes argopy is definitely on my radar and I anticipate that implementing the data fetching using argopy would be a big upgrade for the bgcArgoDMQC package. I'll plan to do so once argopy has the full BGC support like you said, and will focus my efforts developing the package away from any file fetching components.  Thanks!

**Comment by gmaze** [comment]
Great !
I'll keep you posted wrt argopy progress on this
g

**Comment by cgrdn** [comment]
Hi @gmaze - just wanted to touch base. I've been using argopy a lot this year and want to start working on using it within the DMQC package. As far as I know you can get BGC variables using argopy if you're in `mode='expert'` and using `localftp` as your source, but not in the main package at this point?

Based on that I have created a branch where I will start playing with it (https://github.com/ArgoCanada/bgcArgoDMQC/tree/integrate-argopy) - I anticipate I will use it mostly by immediately piping the data to a dataframe as most things work using that already. Just wanted to put it on your radar, no action required by you.

**Comment by gmaze** [comment]
hi @cgrdn 
to give you an update, with the v0.1.14 argopy release, we're starting to roll out support for BGC variables.
you can fetch all BGC data (as in synthetic files) from the Ifremer erddap, it's all explain in the doc https://argopy.readthedocs.io/
